### PR TITLE
OSOE-47: Open-source Data Tables ⭐

### PR DESCRIPTION
**Pull request by sarahelsaig**
OSOE-47

### CODE REVIEW

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

File: test/Lombiq.OSOCE.Tests.UI/Lombiq.OSOCE.Tests.UI.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+      <TargetFramework>net5.0</TargetFramework>
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/Lombiq.OSOCE.Tests.UI.csproj]: Let's keep everything 3.1 until we switch to 6.0 (also, 3.1 will remain supported longer than 5.0).

File: test/Lombiq.OSOCE.Tests.UI/Tests/RetrievalDataTablesTests.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        public Task RecipeDataShouldBeDisplayedCorrectly(Browser browser) =>
+            ExecuteTestAfterSetupAsync(
+                async context =>
+                {
+                    await context.SignInDirectlyAsync();
+                    await context.ExecuteDataTablesSampleRecipeDirectlyAsync();
+
+                    await context.TestDataTableTagHelperAsync();
+                    await context.TestDataTableProviderWithShapeAsync();
+                    await context.TestDataTableIndexBasedProviderAsync();
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/Tests/RetrievalDataTablesTests.cs]: While it's good to have granularity, if necessary, all of this should be available as a single method too, and used here.

File: test/Lombiq.OSOCE.Tests.UI/UITestBase.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+using Xunit.Abstractions;
+
+namespace Lombiq.OSOCE.Tests.UI
+{
+    public class UITestBase : OrchardCoreUITestBase
+    {
+        protected override string AppAssemblyPath => WebAppConfigHelper
+            .GetAbsoluteApplicationAssemblyPath("Lombiq.OSOCE.Web", "netcoreapp3.1");
+
+        protected override Size MobileBrowserSize { get; } = new(414, 736);
```
Comment (Piedone) [inline test/Lombiq.OSOCE.Tests.UI/UITestBase.cs]: Why exactly this resolution?

**Review by Piedone — CHANGES_REQUESTED** [review]
Test is failing.

File: test/Lombiq.OSOCE.Tests.UI/UITestBase.cs (hunk 3 above)
Comment (sarahelsaig) [reply test/Lombiq.OSOCE.Tests.UI/UITestBase.cs]: I copied it from SNOW. Actually this value doesn't need to be overridden so I can remove it.

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Pull request merged**